### PR TITLE
When a code action returns a workspace edit affecting a single file, …

### PR DESCRIPTION
**Pull request by vrubezhny**
…open that file and select the change #543

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java [hunk 1]
```diff
@@ -944,12 +968,55 @@ public static CompositeChange toCompositeChange(WorkspaceEdit wsEdit, String nam
...
 	}
 
+	private static final Range DEFAULT_RANGE = new Range(new Position(0, 0), new Position(0, 0));
+	/*
+	 * Reports the URI and the start range of the given text edit, if exists.
+	 *
+	 *
+	 * @param textEdits A list of textEdits sorted in reversed order
+	 */
+	private static void collectChangedURI(URI uri, List<TextEdit> textEdits, Map<URI, Range> collector) {
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:985]: What about returning the collector instead?

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java [hunk 2]
```diff
@@ -859,6 +871,17 @@ public static void applyWorkspaceEdit(WorkspaceEdit wsEdit, String label) {
...
 	 */
 	public static CompositeChange toCompositeChange(WorkspaceEdit wsEdit, String name) {
+		return toCompositeChange(wsEdit, name, null);
+	}
+
+	/**
+	 * Returns a ltk {@link CompositeChange} from a lsp {@link WorkspaceEdit}.
+	 *
+	 * @param wsEdit
+	 * @param name
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:881]: Missing note about the `collector`.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java [hunk 3]
```diff
@@ -859,6 +871,17 @@ public static void applyWorkspaceEdit(WorkspaceEdit wsEdit, String label) {
...
+	 * @return a ltk {@link CompositeChange} from a lsp {@link WorkspaceEdit}.
+	 */
+	public static CompositeChange toCompositeChange(WorkspaceEdit wsEdit, String name, Map<URI, Range> collector) {
```
Comment (mickaelistria) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:884]: Can't this one be private?

**Review by mickaelistria — CHANGES_REQUESTED** [review]

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java (hunk 3 above)
Comment (vrubezhny) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java]: Yes, it can. Fixed

**Review by mickaelistria — APPROVED** [review]

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java (hunk 2 above)
Comment (vrubezhny) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java:881]: Added `@param collector` param comment

**Comment by mickaelistria** [comment]
Thank you!

**Pull request merged**